rmf_task/src/rmf_task/agv/TaskPlanner.cpp

```diff
@@ -631,8 +646,19 @@ bool Filter::ignore(const Node& node)
 const rmf_traffic::Duration segmentation_threshold =
     rmf_traffic::time::from_seconds(1.0);
 
-} // anonymous namespace
+inline double compute_g_assignment(const TaskPlanner::Assignment& assignment)
```

Comment:
Is there a reason for separating this block of code from `compute_g()`?

Reply:
It was to enable `compute_cost` and `compute_g` to use the same common function. If not, there is some duplication because `compute_cost` takes in an `Assignment` while `compute_g` takes in a `Node` that contains `AssignmentWrapper`s, so there becomes 2 separate functions to keep in sync with each other.

Reply:
I was thinking `compute_g(Node)` can create `Assignments` from the `Node` and then call `compute_cost(Assignments)` on it. But this approach is valid too.